scintilla/gtk/ScintillaGTK.cxx

```diff
@@ -264,6 +279,12 @@ ScintillaGTK::~ScintillaGTK() {
 	}
 	ClearPrimarySelection();
 	wPreedit.Destroy();
+	if (settingsHandlerId) {
+		g_signal_handler_disconnect(settings, settingsHandlerId);
+	}
+	if (settings) {
+		g_object_unref(settings);
```

Comment:
Fairly sure this is the problem. `settings` is initialized using `gtk_settings_get_default()` and this is documented as "transfer-none", so `g_object_unref()` is inappropriate. Code using this instance is then suffering from use-after-free.

Reply:
https://sourceforge.net/p/scintilla/code/ci/db178448ae7a1049853c8e6e2cdba6f798ec1ed5/

I guess we can simply remove the unref as the same fix is alredy applied upstream.